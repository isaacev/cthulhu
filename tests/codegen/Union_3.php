<?php

namespace Union_3 {
  abstract class Foo {}
  class Bar extends \Union_3\Foo {
    public $integer;
    function __construct($a) {
      $this->integer = $a["integer"];
    }
  }
  function main() {
    new \Union_3\Bar([
      "integer" => 123
    ]);
  }
}

namespace {
  \Union_3\main(null);
}
