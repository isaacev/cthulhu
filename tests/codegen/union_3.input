type Foo =
  | Bar { integer: ::Kernel::Int };

#[entry]
fn main() -> () {
  Foo::Bar{ integer: 123 };
}
