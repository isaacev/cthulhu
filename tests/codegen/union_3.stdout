<?php

namespace Kernel {
  // #[inline]
  // #[construct]
  function _print($a) {
    print($a);
  }
  // #[inline]
  function strval($a) {
    return \strval($a);
  }
  // #[inline]
  function mt_rand($a, $b) {
    return \mt_rand($a, $b);
  }
  // #[inline]
  // #[construct]
  function subscript($a, $b) {
    return $a[$b];
  }
}

namespace union_3 {
  abstract class Foo {}
  class Bar extends \union_3\Foo {
    public $integer;
    function __construct($args) {
      $this->integer = $args["integer"];
    }
  }
  // #[entry]
  function main() {
    new \union_3\Bar([
      "integer" => 123
    ]);
  }
}

namespace {
  \union_3\main();
}
